connect/api/src/main/java/org/apache/kafka/connect/sink/SinkRecord.java

```diff
@@ -57,6 +73,105 @@ public TimestampType timestampType() {
         return timestampType;
     }
 
+    /**
+     * Get the original topic for this sink record, corresponding to the topic of the Kafka record before any
+     * {@link org.apache.kafka.connect.transforms.Transformation Transformation}s were applied. This should be used by
+     * sink tasks for any internal offset tracking purposes (that are reported to the framework via
+     * {@link SinkTask#preCommit(Map)} for instance) rather than {@link #topic()}, in order to be compatible with
+     * transformations that mutate the topic name.
```

Comment:
On top of the `Transformation` link nit, we can simplify some of the language here:
```suggestion
     * Get the original topic for this sink record, before any
     * {@link Transformation transformations} were applied.
     * In order to be compatible with transformations that mutate topic names, this method should be used
     * by sink tasks instead of {@link #topic()} for any internal offset tracking purposes (for instance, reporting offsets to the Connect runtime via
     * {@link SinkTask#preCommit(Map)}).
```